packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -0,0 +1,435 @@
+          },
+        });
+        props.onBlur({
+          target: {
+            id: SelectableSearchInput.getTextInputId(selectablSearchInputId),
+            name: getTextInputName(props.name),
+          },
+        });
+      }
+    },
+    [props.onBlur, selectablSearchInputId, props.name]
+  );
+
+  return (
+    <Constraints.Horizontal max={props.horizontalConstraint}>
+      <Container
+        ref={containerRef}
+        onBlur={handleContainerBlur}
+        data-testid="selectable-search-input-container"
+      >
+        <Constraints.Horizontal max={props.menuHorizontalConstraint}>
+          <SelectableSelect
+            {...props}
+            id={SelectableSearchInput.getDropdownId(selectablSearchInputId)}
+            name={getDropdownName(props.name)}
+            textInputHasFocus={textInputHasFocus}
+            dropdownHasFocus={dropdownHasFocus}
+            handleDropdownFocus={handleDropdownFocus}
+            handleDropdownBlur={handleDropdownBlur}
+            isNewTheme={isNewTheme}
+            textInputRef={textInputRef}
+          />
+        </Constraints.Horizontal>
+        <div
+          css={[
+            getSelectableSearchInputContainerStyles(props),
+            dropdownHasFocus &&
+              isNewTheme &&
+              css`
+                border-left-color: ${designTokens.borderColorForInputWhenFocused};
+              `,
+          ]}
+        >
+          <input
+            ref={textInputRef}
+            id={SelectableSearchInput.getTextInputId(selectablSearchInputId)}
+            name={getTextInputName(props.name)}
+            type="text"
+            value={searchValue}
+            onChange={handleChange}
+            onBlur={handleTextInputBlur}
+            onFocus={handleTextInputFocus}
+            disabled={props.isDisabled}
+            placeholder={props.placeholder}
+            readOnly={props.isReadOnly}
+            autoFocus={props.isAutofocussed}
+            autoComplete={props.autoComplete}
+            aria-readonly={props.isReadOnly}
+            contentEditable={!props.isReadOnly}
+            css={getSelectableSearchInputStyles(props)}
+            {...filterDataAttributes(props)}
+            /* ARIA */
+            aria-invalid={props['aria-invalid']}
+            aria-errormessage={props['aria-errormessage']}
+            data-testid="selectable-input"
+          />
+          {props.isClearable &&
+            searchValue &&
+            !props.isDisabled &&
+            !props.isReadOnly && (
+              <SecondaryIconButton
+                icon={<CloseIcon size="medium" />}
+                label={'clear-button'}
+                onClick={handleClear}
+                css={getClearIconButtonStyles(props)}
+              />
+            )}
+          <SecondaryIconButton
+            icon={<SearchIcon />}
+            label={'search-button'}
+            onClick={handleSubmit}
+            css={getSearchIconButtonStyles(props)}
+          />
```

Comment:
Perhaps we want this to be configurable. I'm suggesting because maybe we would like to use this component as a base for the `MoneyInput`.